OrbitClientProtos/capture_data.proto

```diff
@@ -157,6 +159,15 @@ message TimerInfo {
   uint64 user_data_key = 10;
   uint64 timeline_hash = 11;
   repeated uint64 registers = 12;
+  Color color = 13;
```

Comment:
We need to be careful about adding additional data to TimerInfo, simply for memory reasons (not during transfer, but in-memory). I think this "only" adds a pointer to a TimerInfo, but we already have a different way to handle colors. The manual instrumentation timers also have a color that can be customized.

Reply:
@pierricgimmig How are colors handled for manual instrumentation markers?

Reply:
For manual instrumentation Colors are encoded as part of the `orbit_api::EncodedEvent`, which is encoded using `TimerInfo`'s register fields. You may want to have a look at `ThreadTrack::GetUserColor`. 
To be honest, I have high doubts in also using this for the markers here. The new `Color` field should cost no memory if unused and if we using it, it should be still better than using the 6 64bit registers for encoding 4 float values.

Reply:
Also, I guess mid-term we really want to split the TimerInfo into different protos for each test.

Reply:
> The new Color field should cost no memory if unused

It's a pointer in the message, so 8 bytes per TimerInfo, even if the TimerInfo does not use color.

Reply:
>>The new Color field should cost no memory if unused

>It's a pointer in the message, so 8 bytes per TimerInfo, even if the TimerInfo does not use color.

Oh, I thought there was some magic going on, such that unused fields do not use memory at all (even for primitive types). But nevertheless, my argument would be to make it clean and then start separating the timers rather sooner then later.

Reply:
Fields don't use space in encoded messages if they are not present, but the generated code has to have a field for them. You can look at the generated code easily (in the build folder). 

I'm not convinced we should have separate messages for all types, but let's have that discussion when we get there. 

Adding the color message is not going to bloat memory too much, relatively speaking, but I want to make sure we consider changes carefully. 

Another thing to consider here is the serialized size (TimerInfo are used in the serialized capture file). floats are always stored as 4 bytes in encoded format, so a capture with a lot of GPU timers could potentially be a lot larger than necessary. How would you feel about storing the colors as 4 int values and just use values [0, 255]?

Reply:
Storing the values as ints [0,255] would be totally fine for me (can't we even use uint8 here?).

Also agree that we should discuss the rest in detail, when it is the time for this.

Reply:
Done!

Reply:
There is no uint8/int8 type in protos. But for the encoded format that doesn't matter.

Reply:
Right, also noticed this. I used uint32. Should be fine though.

Reply:
@ronaldfw is there anything else I should take care of?